File: app/src/main/java/glnative/example/NativeEglExample.java
package glnative.example;

import android.app.Activity;
import android.os.Bundle;
import android.widget.Toast;
import android.view.Surface;
import android.view.SurfaceView;
import android.view.SurfaceHolder;
import android.view.View;
import android.view.View.OnClickListener;
import android.util.Log;


public class NativeEglExample extends Activity implements SurfaceHolder.Callback
{

    private static String TAG = "EglSample";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Log.i(TAG, "onCreate()");

        setContentView(R.layout.activity_main);
        SurfaceView surfaceView = (SurfaceView)findViewById(R.id.surfaceview);
        surfaceView.getHolder().addCallback(this);
        surfaceView.setOnClickListener(new OnClickListener() {
            public void onClick(View view) {
                Toast toast = Toast.makeText(NativeEglExample.this,
                        "This demo combines Java UI and native EGL + OpenGL renderer",
                        Toast.LENGTH_LONG);
                toast.show();
            }});
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "onStart()");
        nativeOnStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "onResume()");
        nativeOnResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "onPause()");
        nativeOnPause();
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG, "onStop()");
        nativeOnStop();
    }

    public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
        nativeSetSurface(holder.getSurface());
    }

    public void surfaceCreated(SurfaceHolder holder) {
    }

    public void surfaceDestroyed(SurfaceHolder holder) {
        nativeSetSurface(null);
    }


    public static native void nativeOnStart();
    public static native void nativeOnResume();
    public static native void nativeOnPause();
    public static native void nativeOnStop();
    public static native void nativeSetSurface(Surface surface);

    static {
        System.loadLibrary("nativeegl");
    }

}

File: app/src/main/java/glnative/example/NativeEglExample.kt
package glnative.example

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.widget.RelativeLayout
import android.widget.Toast


class NativeEglExample : Activity() {

    var n: NativeView? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Log.i("Assets", "copying assets folder")
        AssetsManager().copyAssetFolder(assets, filesDir.absolutePath + "/ASSETS")

        if (n == null) n = NativeView(applicationContext)

        Log.i(n!!.TAG, "onCreate()")

        // build layout
        val rel = RelativeLayout(applicationContext)
        rel.addView(n!!.surfaceView!!)

        // DO NOT add text!, adding text will incure screen burn when displayed for long periods of time

        // set layout
        setContentView(rel)

        n!!.surfaceView!!.setOnClickListener {
            val toast = Toast.makeText(this@NativeEglExample,
                    "This demo combines Java UI and native EGL + OpenGL renderer",
                    Toast.LENGTH_LONG)
            toast.show()
        }
    }

    override fun onStart() {
        super.onStart()
        Log.i(n!!.TAG, "onStart()")
        n!!.nativeOnStart(filesDir.absolutePath + "/ASSETS", applicationInfo.nativeLibraryDir)
    }

    override fun onStop() {
        super.onStop()
        Log.i(n!!.TAG, "onStop()")
        n!!.nativeOnStop()
    }
}
